server/controllers/upload-controller.js

```diff
@@ -0,0 +1,312 @@
+      })
+      .from("semester")
+      .where("sid", semesterId);
+    if (semester[0].inserted_lectures === 1) {
+      throw {
+        msg: `Already inserted the lecture for Semester ` + semesterId,
+        status: 504,
+      };
+    }
+    await readFile(userId, path, "schedule", semesterId);
+    //console.log("STEP 4 DOPO READFILE (FINE)")
+    await knex("semester")
+      .update({ inserted_lectures: 1 })
+      .where("sid", semesterId); //no more insert of lecture for the selected semester
+    res.status(200).send({
+      //tot_insert: tot,
+      message: "Uploaded the file successfully: " + req.file.originalname,
+    });
+  } catch (error) {
+    console.log(error);
+    res.status(error.status).send({
+      message: error.msg,
+    });
+  }
+  await clearFile(path);
+};
+
+const generateLectures = (semesterId, lecture) => {
+  return new Promise(async (resolve, reject) => {
+    try {
+      const semester = await knex
+        .select({
+          start: "start",
+          end: "end",
+        })
+        .from("semester")
+        .where("sid", semesterId);
+      let semesterDate = semester[0];
+
+      const course = await knex
+        .select({ main_prof: "main_prof" })
+        .from("course")
+        .where("id", lecture.id);
+      let courseProf = course[0];
+      const days = ["Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"];
+      const regex =
+        "^([0-1]?[0-9]|2[0-3]):[0-5][0-9]-([0-1]?[0-9]|2[0-3]):[0-5][0-9]$";
+      if (lecture.time.match(regex) === null || !days.includes(lecture.day)) {
+        resolve(0);
+      }
+      let start = moment(semesterDate.start);
+      let end = moment(semesterDate.end);
+      let arr = [];
+      let tmp = start.clone().day(lecture.day);
+      if (tmp.isSameOrAfter(start, "d")) {
+        //check if the day of the first week is after the start date, if true add a date
+
+        arr.push(createLecture(lecture, courseProf, tmp));
+      }
+      while (tmp.add(7, "days").isBefore(end)) {
+        //generate all days until the end date adding 7 days each time
+
+        arr.push(createLecture(lecture, courseProf, tmp));
+      }
+      //console.log("STEP 1n DOPO AVER CREATO IL GRUPPO DI LECTURE")
+      await knex("lecture").insert(arr);
+      resolve(arr.length);
+    } catch (error) {
+      reject({
+        msg: `There was an error`,
+        status: 503,
+      });
+      //throw { msg: `There was an error`, status: 503 };
+    }
+  });
+};
+
+const createLecture = (lecture, courseProf, date) => {
+  let row = {};
+  row.course = lecture.id;
+  row.lecturer = courseProf.main_prof;
+  row.capacity = lecture.capacity;
+  row.status = "presence";
+  row.room = lecture.room;
+  let times = lecture.time.split("-");
+  let [hour, minute] = times[0].split(":");
+  date.set({ hour, minute });
+  row.start = date.format("YYYY-MM-DD HH:mm:ss");
+  [hour, minute] = times[1].split(":");
+  date.set({ hour, minute });
+  row.end = date.format("YYYY-MM-DD HH:mm:ss");
```

Comment:
Wouldn't it be easier, and also more readable, if you used moment.js? 
Example:
```js
row.start = moment(lecture.time, INPUT_FORMAT).format(FINAL_FORMAT);

```